src/app/legacy/components/Footer/index.jsx

```diff
@@ -70,24 +70,49 @@ const StyledAmpCookieSettingsButton = styled(AmpCookieSettingsButton)`
   }
 `;
 
+const openPrivacyManagerModal = e => {
+  e.preventDefault();
+  if (window.dotcom && window.dotcom.openPrivacyManagerModal) {
+    window.dotcom.openPrivacyManagerModal();
+  }
+};
+
 const SitewideLinks = ({
   links,
   trustProjectLink,
   copyrightText,
   externalLink,
   isAmp,
+  showAdsBasedOnLocation,
   script,
   service,
 }) => {
   const elements = links.map(({ id, text, href, lang }) => {
-    if (isAmp && id === 'COOKIE_SETTINGS') {
-      return (
-        <StyledAmpCookieSettingsButton lang={lang} service={service}>
-          {text}
-        </StyledAmpCookieSettingsButton>
-      );
+    if (id === 'COOKIE_SETTINGS') {
+      if (isAmp) {
+        return (
+          <StyledAmpCookieSettingsButton lang={lang} service={service}>
+            {text}
+          </StyledAmpCookieSettingsButton>
+        );
+      }
+
+      if (showAdsBasedOnLocation) {
```

Comment:
Sorry to be pedantic! But is it possible to use the `onClient()` function here in tandem with the `showAdsBasedOnLocation` check, like `(showAdsBasedOnLocation && onClient())`?

This should allow you to remove all of the `useEffect` stuff from the `Link` component. I tried locally and it displays when JS is enabled and doesn't when its disabled. But I may be missing something!

We try to minimise as many side-effects as possible, so just want to be sure that this is the only approach to the problem here.

Reply:
No problem. I'll give it a go tomorrow...

Reply:
Just had a look and you get a hydration error with that approach:
`Uncaught Error: Hydration failed because the initial UI does not match what was rendered on the server`

Which would you prefer @amoore108 ?

Reply:
Ah good spot, I must have had my dev tools filtering out errors. Yea I think your implementation is ok here then 👍